src/main/java/core/basesyntax/service/impl/DataTransactionParserImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.DataTransactionParser;
+import core.basesyntax.strategy.OperationStrategy;
+import core.basesyntax.strategy.operation.OperationHandler;
+import java.util.Map;
+
+public class DataTransactionParserImpl implements DataTransactionParser {
+    private static final String COMMA = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int FRUIT_NAME_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private final OperationStrategy operationStrategy;
+
+    public DataTransactionParserImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public Map<String, Integer> parseDataTransaction(String data) {
```

Comment:
if we just `parse` data, we shouldn't change state of our `FRUIT_STORAGE`
let's move logic of changing state to `FruitService`, and this method can return list of transactions
like `List< FruitTransaction>`